pkg/sidecars/tshark/tshark_test.go

```diff
@@ -169,7 +169,16 @@ func TestTsharkValidateConfig(t *testing.T) {
 }
 
 func TestTsharkClone(t *testing.T) {
```

Comment:
This update demostrates that our testing is too interwoven. Updates to an `instance.go` file should result in updates to tests in an `instance_test.go` file. Needing to update specific side car tests like tshark to test core functionality is not good. 

Not a blocker for this PR but something we should work towards with this repo. This might require a good amount of mocking architecture to get us to the end goal. 

I think it would be good to resolve this comment into an issue to capture the context/example and then build our an epic from there.

Reply:
Put another way, this repo doesn't have true unit testing, only integration testing.

Reply:
The main reason that the sidecar "unittest" is updated, is because we updated the instance state check and obviously it has to reject some states which was working before without having those checks. Therefore, we need to update the unittests/e2e tests even some code from external users who try to clone an instance which is not `Committed` nor `Stopped`.

I agree that we do not have proper unit tests. There have been some attempts and we have some unit tests for k8s with acceptable coverage rate. But the instance pkg does not have unit test at the moment.